azurelinuxagent/ga/exthandlers.py

```diff
@@ -783,6 +783,19 @@ def __init__(self, ext_handler, protocol):
             self.logger.error(u"Failed to create extension log dir: {0}", e)
 
         log_file = os.path.join(self.get_log_dir(), "CommandExecution.log")
+
+        try:
+            if os.stat(log_file).st_size > execution_log_max_size:
+                with open(log_file, "rb") as existing_log_file_buffer:
+                    existing_log_file_buffer.seek(-1 * execution_log_max_size, 2)
+                    _ = existing_log_file_buffer.readline()
```

Comment:
what was the conclusion wrt utf-8 encoding and truncating a char that has been encoded using more than 1 byte?

Reply:
It seems we are ascii encoding after all-- @larohra pointed out that the comment I thought referred to encoding as utf-8 was actually talking about the python2 unicode object; later on, we encode that object with ascii. However, `seek` and `readline` should both be utf-8 aware anyway.
Truncating a multi-byte character halfway through would cause file corruption, and is therefore something we really want to avoid. As far as I can tell, this is mainly a problem if `read` (not `readline`) is called on a utf-8 encoded file opened in binary mode.

Reply:
ok